### PR TITLE
Add recipe for fastapi-third-party-auth.

### PR DESCRIPTION
**Pull request by antarcticrainforest**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by antarcticrainforest** [comment]
@conda-forge/help-python, ready for review!

**Comment by antarcticrainforest** [comment]
@conda-forge-admin, please ping team

File: recipes/fastapi-third-party-auth/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - requests >=2.24.0
+    - python-jose >=3.2.0
+
+test:
+  imports:
+    - fastapi_third_party_auth
+  commands:
+    - pip check
+  requires:
+    - python >={{ python_min }}
```
Comment (synapticarbors) [inline recipes/fastapi-third-party-auth/meta.yaml:38]:
```suggestion
    - python {{ python_min }}
```

File: recipes/fastapi-third-party-auth/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  sha256: 01afc17ca5c6b0a80782805aa5cfdedbae3b26db52d47e1a5d87a50e9e1ae44a
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >={{ python_min }}
```
Comment (synapticarbors) [inline recipes/fastapi-third-party-auth/meta.yaml:20]:
```suggestion
    - python {{ python_min }}
```

File: recipes/fastapi-third-party-auth/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{% set python_min = "3.8" %}
```
Comment (synapticarbors) [inline recipes/fastapi-third-party-auth/meta.yaml:1]: Unless there is a compelling reason, you should stick with the `python_min` conda-forge uses globally (3.9) unless you require a newer version. If 3.9 is ok, then delete this line.

File: recipes/fastapi-third-party-auth/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    - poetry >=0.12
+    - poetry-core
+    - pip
+  run:
+    - python >={{ python_min }}
+    - fastapi >=0.61.0
+    - pydantic >=1.6.1
+    - cachetools >=4.1.1
+    - requests >=2.24.0
+    - python-jose >=3.2.0
```
Comment (synapticarbors) [inline recipes/fastapi-third-party-auth/meta.yaml:30]: `python-jose` upstream adds an extra: `python-jose = {extras = ["cryptography"], version = ">= 3.2.0"}`. That extra pulls in `cryptography >=3.4.0`:
https://github.com/mpdavis/python-jose/blob/master/setup.cfg#L45C5-L45C25

So I would add that here as an additional requirement for `run`.

**Review by synapticarbors — CHANGES_REQUESTED** [review]

File: recipes/fastapi-third-party-auth/meta.yaml (hunk 3 above)
Comment (antarcticrainforest) [reply recipes/fastapi-third-party-auth/meta.yaml]: Ok, thanks for the clarification.

File: recipes/fastapi-third-party-auth/meta.yaml (hunk 1 above)
Comment (antarcticrainforest) [reply recipes/fastapi-third-party-auth/meta.yaml]: Done

**Pull request merged**